src/k2/httpproxy/HTTPProxy.cpp

```diff
@@ -448,16 +448,19 @@ seastar::future<nlohmann::json> HTTPProxy::_handleQuery(nlohmann::json&& jsonReq
         jsonReq.at("txnID").get_to(txnID);
         jsonReq.at("queryID").get_to(queryID); 
     } catch(...) {
-        return JsonResponse(Status{400, "Bad json for query request"});
+        return JsonResponse(Statuses::S400_Bad_Request("Bad json for query request"));
     }
-    if (_txns.find(txnID) == _txns.end()) {
-        return JsonResponse(Status{400, "Could not find txnID for query request"});
+    auto txnIter = _txns.find(txnID);
+    if (txnIter == _txns.end()) {
+        return JsonResponse(Statuses::S400_Bad_Request("Could not find txnID for query request"));
     }
-    if (_queries.find(queryID) == _queries.end()) {
-        return JsonResponse(Status{400, "Could not find queryID for query request"});
+    auto queryIter = _queries.find(queryID);
```

Comment:
you don't need queryIter so why have a variable for it? You can evaluate find() as before in the if condition to make the code a bit easier on the eyes.

Reply:
I need queryiter  for the query api 6 lines below

https://github.com/futurewei-cloud/chogori-platform/blob/09605d89fbe5ea398e49d338a6f6f1e66f5b9901/src/k2/httpproxy/HTTPProxy.cpp#L463

Reply:
oh yeah, you're right. We should probably combine the two maps when we add support for timeout, but for now, looks good.